### ISSUE TITLE
New solver should prefer newer versions

### ISSUE THREAD

**Issue by ilanschnell** [conda/conda#2197]
@mcg1969 I just tried this (Using conda 4.0.1 on OSX):
```
$ conda create -n foo accelerate python=2.7
Fetching package metadata: ....
Solving package specifications: ...........
...
The following NEW packages will be INSTALLED:

    accelerate: 1.0.1-np16py27_p0 
    bitarray:   0.8.1-py27_0      
    libnvvm:    1.0-p0            
    llvm:       3.2-0             
    llvmpy:     0.11.1-py27_0     
    meta:       0.4.2.dev-py27_0  
    nose:       1.3.7-py27_0      
    numba:      0.7.1-np16py27_0  
    numbapro:   0.10.1-np16py27_p0
    numpy:      1.6.2-py27_4      
    openssl:    1.0.2g-0          
    python:     2.7.11-0          
    readline:   6.2-2             
    sqlite:     3.9.2-0           
    tk:         8.5.18-0          
    zlib:       1.2.8-0           
```
However, when I say:
```
$ conda create -n foo accelerate=2.0.2 python=2.7
```
all is fine. This means that accelerate 2.0.2 is feasible, but for whatever reason the solver prefers this really old accelerate 1.0.1 version. I would expect what when you don't specify a version of a package, that you *always* want the latest.

**Comment by ilanschnell** [comment]
Travis was able to confirm this behavior: https://github.com/conda/conda/issues/2197

**Comment by mcg1969** [comment]
This was a bug in lower-level SAT clause generation. Fix here: https://github.com/conda/conda/pull/2199

**Comment by jjhelmus** [comment]
This issue causes big trouble when a package is removed from a environment during `conda build` as the resulting environment may no long meet the original specs.  For example:

```
$ conda build pip --python=3.5
Removing old build environment
Removing old work directory
BUILD START: pip-8.1.0-py35_99
Fetching package metadata: ....
Solving package specifications: ...........

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    setuptools-20.2.2          |           py35_0         458 KB
    pip-8.1.0                  |           py35_0         1.6 MB
    ------------------------------------------------------------
                                           Total:         2.0 MB

The following NEW packages will be INSTALLED:

    openssl:    1.0.2g-0
    pip:        8.1.0-py35_0
    python:     3.5.1-0
    readline:   6.2-2
    setuptools: 20.2.2-py35_0
    sqlite:     3.9.2-0
    tk:         8.5.18-0
    wheel:      0.29.0-py35_0
    xz:         5.0.5-1
    zlib:       1.2.8-0

Fetching packages ...
setuptools-20. 100% |################################| Time: 0:00:00 948.48 kB/s
pip-8.1.0-py35 100% |################################| Time: 0:00:01   1.43 MB/s
Extracting packages ...
[      COMPLETE      ]|###################################################| 100%
Linking packages ...
[      COMPLETE      ]|###################################################| 100%
pip is installed as a build dependency. Removing.
Solving package specifications: ...........

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    python-1.0.1               |                0         349 KB

The following packages will be REMOVED:

    pip:        8.1.0-py35_0
    setuptools: 20.2.2-py35_0
    wheel:      0.29.0-py35_0

The following packages will be DOWNGRADED:

    python:     3.5.1-0       --> 1.0.1-0
...
```

FYI, pip cannot be installed in Python 1.0.1.

**Comment by mcg1969** [comment]
This isn't actually the same bug. Can you submit a separate issue? Pip should not be removed, because Python claims it as a dependency (whether it should or not)

**Comment by mcg1969** [comment]
This does imply, to be frank, that you can't easily build pip with conda-build.

**Comment by ilanschnell** [comment]
We can still build `pip` (I think), by using (in `~/.condarc`) the setting:
```
add_pip_as_python_dependency: False
```

**Comment by mcg1969** [comment]
I don't see how that's possible if pip is actually in the metadata for python as a dependency.

**Comment by mcg1969** [comment]
Ah, I see, it's added programmatically.

**Comment by jjhelmus** [comment]
> We can still build pip (I think), by using (in ~/.condarc) the setting:
> 
> add_pip_as_python_dependency: False

Did not know that this setting existed, this solves the issue but the removal of packages.  Is this option documented anywhere? 

I am still seeing a change in what packages conda chooses to remove between version 3.19 vs 4.0. `conda build pip python=3.5` does not downgrade Python in conda 3.19.  Not sure if this is a different bug, if so I'm happy to open a new issue.

**Comment by kalefranz** [comment]
Should be fixed with today's `4.0.2` release.
